shared/middleware/files_api.rb

```diff
@@ -498,14 +504,16 @@ def files_put_file(encrypted_channel_id, filename, body)
       manifest_is_unchanged = false
     end
 
-    # if we're also deleting a file (on rename), remove it from the manifest
-    if params['delete']
+    # Handle any deletion request, unless a case-only rename is occuring.
+    if params['delete'] && params['delete'].downcase != downcased_filename
       manifest_delete_comparison_filename = CGI.unescape(params['delete']).downcase
       reject_result = manifest.reject! {|e| e['filename'].downcase == manifest_delete_comparison_filename}
       manifest_is_unchanged = false unless reject_result.nil?
+
+      bucket.delete(encrypted_channel_id, params['delete'].downcase) if params['delete']
```

Comment:
Note that I'm moving the actual deletion before the actual manifest update. This matters should an exception happen somewhere. The change seems better, as I would expect the manifest update to be before or after the other changes, not in the middle of them. Do we care about consistency of the "system" on an exception?